### ISSUE TITLE
Some WebAudio signatures use number[] instead of Float32Array

### ISSUE THREAD

**Issue by iccir** [Microsoft/TypeScript#23160]
**TypeScript Version:**  2.8.1

**Search Terms:** WebAudio, setValueCurveAtTime

**Code**

```ts
let context: AudioContext;

function demo(conext: AudioContext) {
    let gainNode = context.createGain();
    let sampleCount = Math.floor(0.25 * context.sampleRate);

    let values = new Float32Array(100);
    for (let i = 0; i < sampleCount; i++) {
        values[i] = i / sampleCount;
    }

    gainNode.gain.setValueCurveAtTime(values, 0, sampleCount / context.sampleRate);
}
```

**Expected behavior:**
GainNode's first parameter should be a `Float32Array` per the WebAudio specification. TypeScript's `lib.d.ts` file thinks it's a `number[]`. The above code shouldn't warn.

**Actual behavior:**
```Argument of type 'Float32Array' is not assignable to parameter of type 'number[]'```

**Playground Link:** http://www.typescriptlang.org/play/#src=%2F%2F%20WebAudio%20setValueCurveAtTime%20issue%0D%0A%0D%0Alet%20context%3A%20AudioContext%3B%0D%0A%0D%0Afunction%20demo(conext%3A%20AudioContext)%20%7B%0D%0A%20%20%20%20let%20gainNode%20%3D%20context.createGain()%3B%0D%0A%20%20%20%20let%20sampleCount%20%3D%20Math.floor(0.25%20*%20context.sampleRate)%3B%0D%0A%0D%0A%20%20%20%20let%20values%20%3D%20new%20Float32Array(100)%3B%0D%0A%20%20%20%20for%20(let%20i%20%3D%200%3B%20i%20%3C%20sampleCount%3B%20i%2B%2B)%20%7B%0D%0A%20%20%20%20%20%20%20%20values%5Bi%5D%20%3D%20i%20%2F%20sampleCount%3B%0D%0A%20%20%20%20%7D%0D%0A%0D%0A%20%20%20%20gainNode.gain.setValueCurveAtTime(values%2C%200%2C%20sampleCount%20%2F%20context.sampleRate)%3B%0D%0A%7D%0D%0A

**Related Issues:**

**Comment by iccir** [comment]
Additionally, I believe that the following signatures use `number[]` instead of the correct `Float32Array`:

```
interface WaveShaperOptions extends AudioNodeOptions {
    curve?: number[];
    oversample?: OverSampleType;
}

interface PeriodicWaveOptions extends PeriodicWaveConstraints {
    imag?: number[];
    real?: number[];
}
```

However, I'm not able to figure out why `PeriodicWaveOptions` extends `PeriodicWaveConstraints` or how `AudioNodeOptions` is ever used.

**Comment by mhegazy** [comment]
We have mechanisms to override types from the widl file. Look at https://github.com/Microsoft/TSJS-lib-generator/pull/404 as a reference.

**Comment by mhegazy** [comment]
PRs welcomed. You can find more information about contributing lib.d.ts fixes at https://github.com/Microsoft/TypeScript/blob/master/CONTRIBUTING.md#contributing-libdts-fixes.

**Comment by iccir** [comment]
Thanks! I fixed it for `setValueCurveAtTime`. I held off on changing `PeriodicWaveOptions`/`AudioNodeOptions` as I couldn't figure out how they are used.
